File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        List<Integer> luckyNumbers = new ArrayList<>();

        int rows = matrix.length;
        int cols = matrix[0].length;

        int[] minInRow = new int[rows];
        for (int i = 0; i < rows; i++) {
            int minRow = Integer.MAX_VALUE;
            for (int j = 0; j < cols; j++) {
                minRow = Math.min(minRow, matrix[i][j]);
            }
            minInRow[i] = minRow;
        }

        int[] maxInCol = new int[cols];
        for (int j = 0; j < cols; j++) {
            int maxCol = Integer.MIN_VALUE;
            for (int i = 0; i < rows; i++) {
                maxCol = Math.max(maxCol, matrix[i][j]);
            }
            maxInCol[j] = maxCol;
        }

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                if (matrix[i][j] == minInRow[i] && matrix[i][j] == maxInCol[j]) {
                    luckyNumbers.add(matrix[i][j]);
                }
            }
        }

        return luckyNumbers;
    }
}

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
var luckyNumbers  = function(matrix) {
    const luckyNumber = [];
    let row = matrix.length;
    let col = matrix[0].length;
    const minInRow = []
    for(let i=0;i<row;i++){
        let minRow = Number.MAX_SAFE_INTEGER;;
        for(let j=0;j<col;j++){
            minRow = Math.min(matrix[i][j], minRow);
        }
        minInRow.push(minRow);
    }
    const minInCol = [];
    for(let i=0;i<col;i++){
        let maxCol = Number.MIN_SAFE_INTEGER;;
        for(let j=0;j<row;j++){
            maxCol = Math.max(matrix[j][i],maxCol);
        }
        minInCol.push(maxCol);
    }
    
    for(let i=0;i<row;i++){
        for(let j=0;j<col;j++){
            if(matrix[i][j] == minInRow[i] && matrix[i][j] == minInCol[j]){
                luckyNumber.push(matrix[i][j]);
            }
        }
    }
    return luckyNumber;
};